File: primes.cpp
#include <iostream>
#include <vector>

// Constants
const int range = 1000; // Sets the upper limit of where to find primes

// Prototypes
std::vector<int> createVector(); // Creates the vector with numbers from 0 to range
void primeSeive(std::vector<int>); // Removes all non-primes leaving only primes in the vector

int main() {
	
	// Create initial vector
	std::vector<int> primes = createVector();
	
	// Sort, leaving only primes
	//primeSeive(primes);
	
	// Print vector
	for (int i = 0; i < (primes.size() - 1); i++) {
		std::cout << primes[i] << ", ";
	}
	std::cout << primes[primes.size() - 1] << std::endl;

	//Exit
	std::cin.clear();
    getchar();
    return 0;
}

// Creates the vector with numbers from 0 to range
std::vector<int> createVector() {
	
	std::vector<int> primes;
	
	for (int i = 0; i < range; i++) {
		primes[i] = i + 1;
	}
	
	return primes; // Finish
}

// Removes all non-primes leaving only primes in the vector
void primeSeive(std::vector<int> primes) {
	
	for (int i = 0; i < (range / 2); i++) {
		
		int j = i + 1;
		
		while (j < primes.size()) {
			if (primes[j] % primes[i] == 0) {
				primes.erase(primes.begin() + j);
			} else {
				j++;
			}
		}
	}
	
	return; // Finish
}

File: primes.py
scale = 1000000

def createArray() :
    
    primes = [0] * (scale - 1)
    
    for i in range(2, scale + 1) :
        primes[i - 2] = i
    
    return primes

def primeSeive(primes) :
    for index in range(int(scale / 2)) :

        i = index + 1
        
        while i < len(primes) :
            if primes[i] % primes[index] == 0 :
                del primes[i]
            else :
                i += 1
    return

primes = createArray()

primeSeive(primes)

print(len(primes)/scale)

product = 1.0

for i in primes :
    product *= (i - 1) / i

print(product)